### PR TITLE
[READY] DRAKEBORGS & PLUSHIES

### PR DESCRIPTION
**Pull request by Strazyplus**
### **[Attribution-NonCommercial-ShareAlike 3.0 Unported (CC BY-NC-SA 3.0)](https://creativecommons.org/licenses/by-nc-sa/3.0/)**

1595361337 miz-dee_okay-export

## About The Pull Request
Ported from skyrat https://github.com/Skyrat-SS13/Skyrat13/pull/2863
Commissioned artist: **[deviantart.com/mizartz](deviantart.com/mizartz)**
Vorebellies: @ThighHighs

Though a commission with the talented Mizartz we have created a new subset of borgs for all to have, instead of being a dog the robot is a dragon or in this case a drake. the modules include Medical, Security, Mining, Peace, Engineering, and Janitor variants. in addition of  providing new borgshells we have also included plushie variants that can be obtained via plushie crate or via loadout and come in security and medical. 

janitor
mining
peace
security
engineering
medic
ca2rd

Drakeborg and drake plushies fall under [CC BY-NC-SA 3.0](https://creativecommons.org/licenses/by-nc-sa/3.0/)
Share — copy and redistribute the material in any medium or format, Adapt — remix, transform, and build upon the material, however NonCommercial use — You may not use the material for commercial purposes.

this pr closes #12735 the pr of [Blackmajor's Drake Borg port](https://github.com/Citadel-Station-13/Citadel-Station-13/pull/12735).

## Why It's Good For The Game

Brings new subset of robots for borg players as well cute plushies for people to squeeze.

## Changelog
:cl:
add: Added drakeborgs
add: Added drakeplushies
imageadd: added drakeborg sprites
imageadd: added drakeplushie sprites 
code: changed some code - added drakeplushies to backpack loadout
Removed duplicate voresleeper belly sprites from engdrake & jantidrake.
[CC BY-NC-SA 3.0](https://creativecommons.org/licenses/by-nc-sa/3.0/)
add: Added CC BY-NC-SA 3.0 license details to icon/mob/cyborg
moved drakeborg.dmi to icon/mob/cyborg
/:cl:

### CODE REVIEW

**Comment by ShadeAware** [comment]
THATS MY FUCKIN BOOOOOY 🎉

**Comment by jjpark-kb** [comment]
make sure to have "closes https://github.com/Citadel-Station-13/Citadel-Station-13/pull/12735" in your comment.

**Comment by Strazyplus** [comment]
Heres the drakeborgs, the licence is here, the credit is here, it has vorbellies, it works. now lets get it merged... I been waiting too long for this day to let lack of credit hold back the drakeborgs release.

be nice, go pet a dragon(drake) or be one, else hug a drake plushie. this is why we made this.

**Comment by BlackMajor** [comment]
Put them in widerobot.dmi, there's no need to have their own dedicated .dmi on upstream.

Also, change the sleeper overlay of drakeeng and drakejanit to the sec drakes. There's no need to have three copies of the same sprite, since they all have the same belly.

**Comment by KeRSedChaplain** [comment]
did Ratvar overlays get added to them yet as well

because those are kind of a must even if the mode isn't in rotation 😂

**Comment by KeRSedChaplain** [comment]
offcenter
Also what's the reason for all of these being way off-centered, am I just dumb and should know why

**Comment by BlackMajor** [comment]
> Also what's the reason for all of these being way off-centered, am I just dumb and should know why

They're off centered so they fit in the radial menu.

**Comment by Strazyplus** [comment]
> Put them in widerobot.dmi, there's no need to have their own dedicated .dmi on upstream.
> 
> Also, change the sleeper overlay of drakeeng and drakejanit to the sec drakes. There's no need to have three copies of the same sprite, since they all have the same belly.

There are going to be more additions and changes for the drakeborg in the future, for example the weapons (disabler, Laser, saw) may get a drake variant sprite in the future and including the cover sprites.

The drakemech.dmi contains credit for the original sprites, **attribution stays intact.**

the widerobot.dmi is already large enough but also tends to have animated sprites within to mysteriously turn into one frames

**Comment by KeRSedChaplain** [comment]
makes sense

**Comment by KeRSedChaplain** [comment]
also uh, _I really don't think there needs to be another unused sprite_. Sure okay it gives credit but credit has been given a total of at least three times including the sprite, there's the PR, the code, and the .dmi file

You also could just put the sprite giving credit into the widerobot file _along with the drakeborg sprites?_

**Comment by Strazyplus** [comment]
> Also, change the sleeper overlay of drakeeng and drakejanit to the sec drakes. There's no need to have three copies of the same sprite, since they all have the same belly.

Going to recend although I plan to add more in the future, being trim isnt a bad idea. Janit and Engineer borgs share security vorebellies.

**Comment by MrJWhit** [comment]
Hey, why does this fall under CC BY-NC-SA 3.0? Did the original author claim that this is free for anyone?
Because we are at least vaguely commercial, there are donations and benefits for donating. This isn't a "product" by most traditional viewpoints, but it's not non-commercial.

File: code/modules/mob/living/silicon/robot/robot_modules.dm
Comment (BlackMajor) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]:
```suggestion
```
This line is unnecessary.

**Comment by Strazyplus** [comment]
> Hey, why does this fall under CC BY-NC-SA 3.0? Did the original author claim that this is free for anyone?
> Because we are at least vaguely commercial, there are donations and benefits for donating. This isn't a "product" by most traditional viewpoints, but it's not non-commercial.

You can't use drakeborg or drakeplushies for Commerical gain. this is non-issue since drakeborgs are accessible to everyone and drakeplushies are also via in-game plushie crate and loadout selection. but if they are not they can't be purchased or gained by being a donator. I'd personally have no problems with them being used as donator bonuses however this is the artist's licence for their work.

**Comment by MrJWhit** [comment]
I would argue that "hey we have these sprites" **is** a commercial gain.

File: code/modules/mob/living/silicon/robot/robot_modules.dm
Comment (Strazyplus) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: oops that slipped

**Comment by Strazyplus** [comment]
> I would argue that "hey we have these sprites" **is** a commercial gain.

Unless there's a bitcoin miner hiding in the code, I think it's fine.

**Comment by MrJWhit** [comment]
Is that your opinion as the author, as the commissioned payer, or are you speaking for the artist?

**Comment by Ludox235** [comment]
Poggies.

**Review by silicons — CHANGES_REQUESTED** [review]
everything is still pending considering bhijn is yet to hear back from the artist
regardless, i am in favor of drakeborgs being in a separate file as they are a new class of cyborg sprites. however, they should be in a cyborg icon directory or something, maybe make icons/mob/cyborg/drakes.dmi or something.
there's also another reason to do this as i believe the precedent for differently licensed files on tgcode was to mark the file with licensing information in the directory and separating it out would make this easily able to happen.

credit wise: there is no reason to put the license and everything on 7 places in the code.
put all of this information in a license .txt marking the file you put the sprites in on the folder. maybe a comment at the top of robot_modules.dm saying that drake sprites have different requirements/licensing than the rest of the assets would be acceptable. 

i personally also feel like it would be right to allow navier right of way for this pull request considering they were the original author assuming they are willing to compromise via the above. and of course, if you have any issues with this, talk to me on discord on coderbus @Strazyplus

**Comment by silicons** [comment]
oh yeah and don't bother with modular_citadel, we've been actually trying to get rid of that. plus, putting it as a new file makes it "modular" enough already, don't need to put it in the deprecated folder that'll need moving later anyways.

**Comment by Strazyplus** [comment]
> everything is still pending considering bhijn is yet to hear back from the artist
> regardless, i am in favor of drakeborgs being in a separate file as they are a new class of cyborg sprites. however, they should be in a cyborg icon directory or something, maybe make icons/mob/cyborg/drakes.dmi or something.
> there's also another reason to do this as i believe the precedent for differently licensed files on tgcode was to mark the file with licensing information in the directory and separating it out would make this easily able to happen.
> 
> credit wise: there is no reason to put the license and everything on 7 places in the code.
> put all of this information in a license .txt marking the file you put the sprites in on the folder. maybe a comment at the top of robot_modules.dm saying that drake sprites have different requirements/licensing than the rest of the assets would be acceptable.
> 
> i personally also feel like it would be right to allow navier right of way for this pull request considering they were the original author assuming they are willing to compromise via the above. and of course, if you have any issues with this, talk to me on discord on coderbus @Strazyplus

The license is https://creativecommons.org/licenses/by-sa/3.0/ I spoke to the artist seven hours ago, and this is the license for the drakeborg sprites.

I will move the drakeborg sprites out of citadel modular and as well create the license.txt and put the licensing info on the top in robot_module.dm

**Comment by Tupinambis** [comment]
Drakeborg sprites do not fit in with the theme of ss13 at all. They have no mechanical or "functional" appearance that makes them fit the "job" so to speak. The general shape isn't great, the animation on idle looks very out of place, and similarly to valeborgs these just look more like toys or something rather than a useful cybernetic unit designed to do work.

I'm not wanting to shit on the work, it just looks very out of place.

**Comment by silicons** [comment]
p.s. i should say i do not have an opinion on the cyborg sprites at all and i know quite a few people dislike it, i'm more interested in resolving the deadlock over licensing and attribution, so there's that.

**Comment by LetterN** [comment]
you should probably remove the comments on the code, as the code is strict AGPL-3.0, you can however add a license.txt (for the icon), with a content like this.

for reference: https://github.com/tgstation/tgstation/blob/master/sound/ambience/license.txt
```
icons/obj/plushies.dmi's icon state of secdrake and meddrake by Mizartz. It has been licensed under the CC BY-NC-SA 3.0 license.

CC BY-NC-SA 3.0 https://creativecommons.org/licenses/by-sa/3.0/
```

**Comment by Strazyplus** [comment]
Hows this?

File: code/game/objects/items/license.txt
Comment (LetterN) [inline code/game/objects/items/license.txt]: place this into the icons/ instead of code/game/objects/items

File: code/game/objects/items/plushes.dm
Comment (LetterN) [inline code/game/objects/items/plushes.dm]: delete this, they're already atributed on the license.txt for the _icons_, as they made the icon.

File: code/modules/mob/living/silicon/robot/robot_modules.dm
Comment (LetterN) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: delete this

File: code/game/objects/items/license.txt
Comment (LetterN) [inline code/game/objects/items/license.txt]:
```suggestion
icons/obj/plushies.dmi's icon state of secdrake and meddrake by Mizartz. It has been licensed under the CC BY-NC-SA 3.0 license.
icons/mob/cyborg/drakemech.dmi by Mizartz. It has been licensed under the CC BY-NC-SA 3.0 license.
```
recommend adding that too

**Comment by Detective-Google** [comment]
Going to mirror my feedback.

I don't think these sprites look good in game, I find them incredibly unsettling. I don't think they encourage a desired behavior, and generally they look utterly out of place.

The overwhelming feedback from players both in discord and in game is that these aren't sprites they want added, and I am rather disappointed that the PR was opened and instantly self-testmerged, which to me is a pretty clear indicator that this change is going to get pushed through regardless of the majority's opinion on the sprites because maintainers generally are 'into' the material.

**Comment by Strazyplus** [comment]
> Going to mirror my feedback.
> 
> I don't think these sprites look good in game, I find them incredibly unsettling. I don't think they encourage a desired behavior, and generally they look utterly out of place.
> 
> The overwhelming feedback from players both in discord and in game is that these aren't sprites they want added, and I am rather disappointed that the PR was opened and instantly self-testmerged, which to me is a pretty clear indicator that this change is going to get pushed through regardless of the majority's opinion on the sprites because maintainers generally are 'into' the material.

I wish I knew why you felt unsettled by drakeborgs if you are by the vorebellies, I understand completely... as for player behavior I can only wish for better players to play drakeborgs.

File: code/game/objects/items/license.txt
Comment (Strazyplus) [reply code/game/objects/items/license.txt]: late, will edit morning + was silicons suggestion for comment on top.

**Comment by nik707** [comment]
Yeah I'm with Google here, they just look pretty silly, don't fit the theme, and somehow seem like even more of a dramabait than dogborgs. I'm fairly heavily against them. Fuck they haven't even been added yet and we got 60+ comments of babyrage.

**Comment by IronEleven** [comment]
Resposting my thoughts from the Discord:
I'm concerned about adding another feral bodytype unavailable to organic players to borg, because you get the dogborg problem of people rolling borg because they want to be a feral, not an enslaved robot.

**Review by silicons — DISMISSED** [review]
acceptable but get rid of code/game/obejcts/items/license.txt as the license isn't for the code it's for the art

**Do not take this approval as my support for the *content* of the PR, I am simply okay with the status of the licensing/attribution at this point.** I don't have any personal preference on the former.

**Comment by Strazyplus** [comment]
license.txt moved to icons where it meant to be.
code referencing the sprites are commented only once for credit and reference to the difference of license.

File: modular_citadel/code/modules/client/loadout/backpack.dm
Comment (timothyteakettle) [inline modular_citadel/code/modules/client/loadout/backpack.dm]: personally its better to scrap loadout plushies and make it a choice beacon so people can start with /any/ loadout plushie (apart from those with special effects)

edit: fuck it I can just do this myself but uh yeah, loadout bloat bad, consider removing them

**Comment by timothyteakettle** [comment]
also you still have the entirely useless sprite that isn't used in any way except to give credit, even though credit is already given

File: modular_citadel/code/modules/client/loadout/backpack.dm
Comment (Strazyplus) [reply modular_citadel/code/modules/client/loadout/backpack.dm]: I am a terrible coder, I have never messed with choice beacon code, while that would reduce bloat on loadout list this would, however, prevent people from having the choice of having multiple plushies if that is what they wanted. the plushies are added so they can be obtained.

**Comment by Strazyplus** [comment]
> also you still have the entirely useless sprite that isn't used in any way except to give credit, even though credit is already given

With everyone's input the credit/licence info is trim and minimal, having a single credit providing sprite when opening DMI file is non-issue.

File: code/game/objects/items/license.txt
Comment (Strazyplus) [reply code/game/objects/items/license.txt]: Moved to icons/

File: code/modules/mob/living/silicon/robot/robot_modules.dm
Comment (Strazyplus) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: took silicons suggestion by instead of seven comments just make one mentioning the credit & the licence on the top.

**Comment by CameronWoof** [comment]
plushes.dm and robot_modules.dm are both still tagged with the artist's name
there is still an unused sprite with the artist's signature and Navier's discord tag

**Comment by CameronWoof** [comment]
Still there, still unnecessary bloat. Now comes with demands. 
image

**Comment by Strazyplus** [comment]
> Still there, still unnecessary bloat. Now comes with demands.
> image

What's the problem?

**Comment by CameronWoof** [comment]
Graffiti'ing the codebase with your logo is obnoxious, and if every spriter did that, it would be a complete shitshow mess. 

Why is it so important this artist is enshrined irrevocably in this dmi? What makes them more special than every other person who has dedicated their time to the evolution of this game _free of charge_? They have been accredited in the PR itself, their license is being respected, they have been paid. We do not need to make this ridiculous additional concession. It's tasteless and masturbatory, and we do not need or want it.

**Comment by nik707** [comment]
> Graffiti'ing the codebase with your logo is obnoxious, and if every spriter did that, it would be a complete shitshow mess.
> 
> Why is it so important this artist is enshrined irrevocably in this dmi? What makes them more special than every other person who has dedicated their time to the evolution of this game _free of charge_? They have been accredited in the PR itself, their license is being respected, they have been paid. We do not need to make this ridiculous additional concession. It's tasteless and masturbatory, and we do not need or want it.

this

**Comment by Detective-Google** [comment]
> > Graffiti'ing the codebase with your logo is obnoxious, and if every spriter did that, it would be a complete shitshow mess.
> > Why is it so important this artist is enshrined irrevocably in this dmi? What makes them more special than every other person who has dedicated their time to the evolution of this game _free of charge_? They have been accredited in the PR itself, their license is being respected, they have been paid. We do not need to make this ridiculous additional concession. It's tasteless and masturbatory, and we do not need or want it.
> 
> this

**Comment by deathride58** [comment]
@CameronWoof Some artists require their work be paired with a watermark of some sorts. This is far from an unreasonable ask, as a lot of artists make a living solely off of their artwork. The reason why the vast majority of other contributions lack this sort of treatment is due to those sprites coming directly from the contributors with the full understanding that the sprites are going into a project with an asset license of CC-BY-SA 3.0, with the expectation that they are doing so entirely for free, and with the full knowledge that the git history combined with their presence in the general SS13 community both serve as their primary forms of credit. In this situation, a professional artist from outside of the general SS13 community was given a fairly high-value commission for quite a few animated sprites, and to my knowledge there has been no attempts to negotiate the use terms as per the artist's TOS (for reference, the artist's TOS here explicitly states that the artist is the one who owns the rights to these sprites, that these sprites may not be used for commercial use, and most relevant to this paragraph, that the signature **must** remain attached to their work). Watermarking is a very common and reasonable ask from professional artists, and I'd say that it's very much fair for the artist to ask that the thing they accepted money in exchange for under their ToS remains compliant with their ToS.

That being said though, yeah. If Strazy's saying the truth (which I'm assuming they are), then this license is valid given what they've stated of the artist's interpretation of the NC clause. However, I'm not sure if my email to the artist went through or not. So I guess I'll just fling them a twitter DM and see if that works. That's basically the only thing holding this PR up at this point.

HOWEVER THOUGH! The CC-BY-SA-**NC** bit in particular is uh... Slightly questionable considering the whole, yknow, we-accept-money-in-exchange-for-ingame-items-which-kinda-constitutes-commercial-use-but-yknow-what-goon-doesnt-slap-anyone-for-it-and-it-seems-like-the-artist's-views-are-the-same-as-goon's-here-if-strazy's-forwarded-it-proper-so-i-guess-its-fine-maybe-possibly-maybe thing. It's also worth noting that the NC clause directly conflicts with AGPL, since the GPL requires that it's possible to use the project for commercial use out of the box. But, again, due to what was said of the artist's interpretation of the NC clause, this ***MIGHT??*** technically be GPL-compatible? IANAL but it looks like it's probably fine given the stated interpretation. Since the sprites are quarantined, it should hopefully be fairly easy to just remove them or replace them if any license issues arise as a result of the elephant in the room here, but! I'm assuming good faith here, so it should hopefully be fine? Hopefully.

Again, I've yet to hear back from the artist, but I'll be poking them on twitter and making sure that they've given express consent to having their work be licensed under CC-BY-SA-NC 3.0, and seeing if they'd be willing to write up a less questionably-gpl-compatible license. Word from the artist is the only thing that's holding this PR back, TBH

**Review by deathride58 — DISMISSED** [review]
UPDATE: I have spoke with the artist! They have given express consent for the content here to be licensed under CC-BY-NC-SA 3.0, with the NC clause being under the interpretation that commercial use constitutes monetizing the materials directly (IE, making them a part of donator-exclusive features, or otherwise locking them behind a paywall). This means that they're *teeeeeechnically* GPL-compatible, and also means that it's compatible with the way we handle donations (we don't have to worry about T2 or T3 donations being a thing, since these items/skins are available for everyone)

**Comment by Ghommie** [comment]
Merge Conflicts. `drakemech.dmi` is fine considering their own license on the assets.

**Comment by Strazyplus** [comment]
@Ghommie @deathride58 Merge conflict, what?

**Comment by Strazyplus** [comment]
I do not know what the sudden conflict is but I do not see it. apparently someone added a hairball 7 days ago. I made sure to have the hairball within the plushie.DMI
 but I am unable to view the conflict with the robot_module.dm

**Comment by Strazyplus** [comment]
conflicts are not going away even after updating.

**Comment by Detective-Google** [comment]
Local man complains a maintainer ported his code and insists on doing it himself, doesnt even know how to fix a merge conflict

**Comment by Strazyplus** [comment]
> Local man complains a maintainer ported his code and insists on doing it himself, doesn't even know how to fix a merge conflict

Yeah, I guess I don't. was asking for a suggestion on trying since commits don't resolve and button to resolve is greyed out.  Asking for slight help? ha silly me what a fool.

Honestly, I don't care that he wanted to port, what I do care about was credit for the artist being scrubbed if you didn't know.

Anyway, I will fix these conflicts.

**Comment by Strazyplus** [comment]
Thanks to the one who actually helped, so fucking simple. I am mad.

**Review by deathride58 — APPROVED** [review]
it's good to merge now. thanks!

**Comment by KeRSedChaplain** [comment]
sigh

**Pull request merged**
